server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1047,16 +1071,19 @@ private static void validateCompositeTemplate(final ClusterState state,
                     // shard id and the current timestamp
                     xContentRegistry, tempIndexService.newQueryShardContext(0, null, () -> 0L, null));
 
+                // triggers inclusion of _timestamp field and its validation:
+                String indexName = DataStream.BACKING_INDEX_PREFIX + temporaryIndexName;
```

Comment:
Should we only add the `_timestamp` mapping if this template has a data stream block? Otherwise it could just be a normal template that happens to contain a field named `@timestamp`, and we attempt to validate that mapping.